server/src/test/java/de/pnp/manager/server/ServerTestBase.java

```diff
@@ -0,0 +1,117 @@
+import com.microsoft.playwright.Playwright;
+import de.pnp.manager.component.Universe;
+import de.pnp.manager.server.database.UniverseRepository;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@ContextConfiguration(initializers = ServerTestBase.ServerTestBaseInitializer.class)
+public abstract class ServerTestBase {
+
+    @Autowired
+    private UniverseRepository universeRepository;
+
+    private @Autowired AutowireCapableBeanFactory beanFactory;
+
+    protected @Nullable Browser browser;
+
+    protected TestServerParameters testServerParameters;
+
+    ServerTestBase() {
+        initializeConfiguration();
+    }
+
+    protected int getPort() {
+        return getTestServerAnnotation().value().getPort();
+    }
+
+    public AutowireCapableBeanFactory getBeanFactory() {
+        return beanFactory;
+    }
+
+    private void initializeConfiguration() {
+        ServerTestBaseInitializer.PORT = getPort();
+    }
+
+    @BeforeEach
+    protected void setup() {
+        testServerParameters = getTestServerAnnotation().value().setupTestData(this);
+        if (isUiTestServer()) {
+            startUiServer();
+        }
+    }
+
+    @AfterEach
+    protected void tearDown() {
+        for (Universe universe : universeRepository.getAll()) {
+            universeRepository.remove(universe.getName());
+        }
+        if (isUiTestServer()) {
+            stopUiServer();
+        }
+    }
+
+    private void stopUiServer() {
+        if (browser != null) {
+            browser.close();
+        }
+    }
+
+    private void startUiServer() {
+        Playwright playwright = Playwright.create(new Playwright.CreateOptions());
+        BrowserType.LaunchOptions launchOptions = new BrowserType.LaunchOptions().setHeadless(false);
+        browser = playwright.chromium().launch(launchOptions);
+
+    }
+
+    private boolean isUiTestServer() {
+        Annotation[] declaredAnnotations = getClass().getDeclaredAnnotations();
+        List<UiTestServer> testServerAnnotations = Arrays.stream(declaredAnnotations).filter(UiTestServer.class::isInstance).map(UiTestServer.class::cast).toList();
+        return !testServerAnnotations.isEmpty();
+    }
```

Comment:
You are not checking if one of the base classes has the `UiTestServer` annotation